RevenueCatUI/CustomerCenter/URLUtilities.swift

```diff
@@ -14,19 +14,28 @@
 //
 
 import Foundation
+import SwiftUI
 
 enum URLUtilities {
 
-    static func createMailURL() -> URL? {
-        let subject = "Support Request"
-        let body = "Please describe your issue or question."
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+    @available(macOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(visionOS, unavailable)
+    static func createMailURLIfPossible(email: String, subject: String, body: String) -> URL? {
         let encodedSubject = subject.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? ""
         let encodedBody = body.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? ""
 
-        // swiftlint:disable:next todo
-        // TODO: make configurable
-        let urlString = "mailto:support@revenuecat.com?subject=\(encodedSubject)&body=\(encodedBody)"
-        return URL(string: urlString)
+        let urlString = "mailto:\(email)?subject=\(encodedSubject)&body=\(encodedBody)"
+
+#if os(iOS)
```

Comment:
Hmm I wonder if we should just mark this entire function with this macro... Since this basically only works in iOS. We could also leave a note to revisit this later to support other systems, but since we already have a bunch of these, it might be ok.

Reply:
yeah, good point!